funnel/views/profile.py

```diff
@@ -301,14 +311,34 @@ def edit_logo_url(self):
             template='img_upload_formlayout.html.jinja2',
         )
 
+    @route('remove_logo', methods=['POST'])
+    @render_with(json=True)
+    @requires_login
+    @requires_roles({'admin'})
+    def remove_logo(self):
+        form = self.CsrfForm()
+        if form.validate_on_submit():
+            self.obj.logo_url = None
+            db.session.commit()
+            flash(_("The banner image has been removed from this profile"), 'info')
+            return render_redirect(self.obj.url_for(), code=303)
+        else:
+            current_app.logger.error(
+                "CSRF form validation error when removing profile logo."
+            )
+            flash(_("There was an issue removing the logo. Try again."), 'error')
```

Comment:
Do not use ambiguous statements like "There was an issue". This outcome can happen when the user was tricked into making a submission via XSS, which means they don't know what action was just attempted. The text could be:

"Were you trying to remove the logo? Try again to confirm."

This still needs some tweaking because it contains "you" and the phrase "try again to confirm" sounds wrong.

Reply:
I've used the sentence you mentioned. Any improvement you can think of? We check for CSRF failure through the codebase. We dont really have a uniform error for them.